nautobot/utilities/tests/test_filters.py

```diff
@@ -826,71 +832,106 @@ class Meta:
 
     def test_site_name_negation(self):
         params = {"name__n": ["Site 1"]}
-        self.assertEqual(SiteFilterSet(params, self.site_queryset).qs.count(), 2)
+        self.assertEqual(
+            SiteFilterSet(params, self.site_queryset).qs.count(), Site.objects.exclude(name="Site 1").count()
+        )
 
     def test_site_slug_icontains(self):
         params = {"slug__ic": ["-1"]}
-        self.assertEqual(SiteFilterSet(params, self.site_queryset).qs.count(), 1)
+        self.assertEqual(
+            SiteFilterSet(params, self.site_queryset).qs.count(), Site.objects.filter(slug__icontains="-1").count()
+        )
 
     def test_site_slug_icontains_negation(self):
         params = {"slug__nic": ["-1"]}
-        self.assertEqual(SiteFilterSet(params, self.site_queryset).qs.count(), 2)
+        self.assertEqual(
+            SiteFilterSet(params, self.site_queryset).qs.count(), Site.objects.exclude(slug__icontains="-1").count()
+        )
 
     def test_site_slug_startswith(self):
         params = {"slug__isw": ["abc"]}
```

Comment:
Now that we're using factory-generated test data, a lot of these hard-coded string values aren't going to match *any* instances. Probably need to dynamically select these as well, e.g.:

```suggestion
        startswith = Site.objects.first().slug[:3]
        params = {"slug__isw": [startswith]}
```